### PR TITLE
Add working directory input

### LINKED ISSUE #1: Support for working-directory: parameter
GabrielBB/xvfb-action [supported](https://github.com/GabrielBB/xvfb-action#example-usage) an optional `working-directory:` to run commands in a different directory from the root. This can be important to set for certain commands such as running jupyter notebooks, because `run: cd some-dir` doesn't work. It would be good for headless-gui to support this argument as well! 🙏

### PR DESCRIPTION
**Pull request by aganders3**
This adds support for a `working_directory` input, as requested in #1. 

Implemented here the `working_directory` input sets the working directory only for commands in the `run` section. I'm open to input on whether this should be applied generally and also apply to the setup and teardown commands.

For compatibility with GabrielBB/xvfb-action, this allows the working directory to be specified with either `working-directory` or `working_directory` (preferred). I'd probably rather leave this out if it's not a blocker for adoption.

closes #1

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -20,6 +20,14 @@ inputs:
     description: 'additional command to run on linux after the main section'
     required: false
     default: 'herbstclient quit'
+  working_directory:
```
Comment (potating-potato) [inline action.yml:23]: I would suggest just leaving one, worst case scenario someone may need to make another trivial commit to correct the parameter name

File: src/main.ts [hunk 2]
```diff
@@ -106,12 +116,14 @@ async function main() {
...
     }
   } catch (error) {
-    // TODO: mock this shit for tests
+    // TODO: upload /tmp/linux*output on failure?
     if (error instanceof Error) core.setFailed(error.message);
     if (error instanceof Error) console.log(error.message);
     // attempt to kill any remaining Xvfb - noop if there's an error at this point
     try {
-      await killAllXvfb(9);
+      if (process.platform == "linux") {
```
Comment (potating-potato) [inline src/main.ts:124]: this change seems unrelated so just making sure it is not added by accident. Also double checking on the strict string match, I am not familiar with this variable, is all linux distributions etc. going to be marked as "linux" here so we don't have mismatches?

Comment (aganders3) [reply src/main.ts:119]: Ah yeah it's unrelated but came up in testing. I can put this in a separate bugfix PR instead if you'd like.

According to the docs 'linux' should be all we expect, but maybe should use `===`? Either way this was the test used in the action this replaces when checking whether to run Xvfb or not, so I think it's okay.
https://nodejs.org/api/process.html#processplatform

**Comment by jni** [comment]
> For compatibility with GabrielBB/xvfb-action, this allows the working directory to be specified with either working-directory or working_directory

❤️ 

> @aganders3 [Remove working-directory input, keep only working_directory](https://github.com/aganders3/headless-gui/pull/2/commits/16009cedb22cffcfc8e0ea29729423db4778ad10)

Oh no! 😂 

Anyway, as noted in #1 also, I do think being a drop-in replacement for xvfb-action should be a goal here. We have a big community that is going to need to migrate, and we want to make that as easy as possible.

**Comment by aganders3** [comment]
😂 it's easy enough to put that back in!

**Review by potating-potato — APPROVED** [review]
Thanks for updating the naming conventions for this!

**Review by jni — APPROVED** [review]
❤️ 🙏
